tests/plugins/test_pulp_tag.py

```diff
@@ -0,0 +1,262 @@
+            'x86_64': {
+                'build': {
+                    'build-name': 'build-1-x64_64',
+                },
+                'metadata_fragment': 'configmap/build-1-x86_64-md',
+                'metadata_fragment_key': 'metadata.json',
+            },
+            'ppc64le': {
+                'build': {
+                    'build-name': 'build-1-ppc64le',
+                },
+                'metadata_fragment': 'configmap/build-1-ppc64le-md',
+                'metadata_fragment_key': 'metadata.json',
+            }
+        }
+    }
+
+    workflow.build_result = BuildResult(logs=["docker build log - \u2018 \u2017 \u2019 \n'"],
+                                        image_id="id1234", annotations=annotations)
+    build_info = {}
+    build_info['x86_64'] = BuildInfo()
+    build_info['ppc64le'] = BuildInfo()
+
+    for platform, v1_image_id in v1_image_ids.items():
+        build_info[platform] = BuildInfo(v1_image_id)
+
+    workflow.plugin_workspace = {
+        OrchestrateBuildPlugin.key: {
+            WORKSPACE_KEY_BUILD_INFO: build_info
+        }
+    }
+
+    mock_docker()
+    return tasker, workflow
+
+
+@pytest.mark.skipif(dockpulp is None,
+                    reason='dockpulp module not available')
+@pytest.mark.parametrize(("v1_image_ids", "should_raise"), [
+    ({'x86_64': None, 'ppc64le': None}, False),
+    ({'x86_64': None, 'ppc64le': 'ppc64le_v1_tag'}, False),
+    ({'x86_64': 'ppc64le_v1_tag', 'ppc64le': 'ppc64le_v1_tag'}, True),
+])
+def test_pulp_tag_basic(tmpdir, monkeypatch, v1_image_ids, should_raise, caplog):
+    tasker, workflow = prepare(v1_image_ids)
+    monkeypatch.setenv('SOURCE_SECRET_PATH', str(tmpdir))
+    with open(os.path.join(str(tmpdir), "pulp.cer"), "wt") as cer:
+        cer.write("pulp certificate\n")
+    with open(os.path.join(str(tmpdir), "pulp.key"), "wt") as key:
+        key.write("pulp key\n")
+
+    runner = PostBuildPluginsRunner(tasker, workflow, [{
+        'name': PLUGIN_PULP_TAG_KEY,
+        'args': {
+            'pulp_registry_name': 'test'
+        }}])
+
+    if should_raise:
+        with pytest.raises(PluginFailedException):
+            runner.run()
+        return
+
+    msg = None
+    expected_results = {}
+    for platform, v1_image_id in v1_image_ids.items():
+        if v1_image_id:
+            msg = "tagging v1-image-id {0} for platform {1}".format(v1_image_id, platform)
+            latest_tag = "latest:{0}".format(v1_image_id)
+            strange_tag = "asd:{0}".format(v1_image_id)
+            expected_results = {
+                'redhat-image-name1': {'tag': latest_tag},
+                'redhat-image-name3': {'tag': strange_tag},
+                'redhat-prefix-image-name2': {'tag': latest_tag}
+            }
+            break
+
+    results = runner.run()
+    if msg:
+        assert msg in caplog.text()
+    else:
+        assert "tagging v1-image-id" not in caplog.text()
+    assert results['pulp_tag'] == expected_results
```

Comment:
Oh, I see, because there may be multiple repos.

But in fact, in the test data, we can make there only be one repo, right?